lib/client/workflow-client/index.js

```diff
@@ -0,0 +1,299 @@
+var q = require('q');
+var _ = require('lodash');
+var shortid = require('shortid');
+var CONSTANTS = require('../../constants');
+var config = require('../../config');
+var MediatorTopicUtility = require('fh-wfm-mediator/lib/topics');
+var mediator, manager, workflowSyncSubscribers;
+
+/**
+ *
+ * Getting Promises for done and error topics.
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {*}
+ */
+function getTopicPromises(doneTopicPromise, errorTopicPromise) {
+  var deferred = q.defer();
+
+  doneTopicPromise.then(function(createdWorkflow) {
+    deferred.resolve(createdWorkflow);
+  });
+
+  errorTopicPromise.then(function(error) {
+    deferred.reject(error);
+  });
+
+  return deferred.promise;
+}
+
+
+/**
+ *
+ * Creating a new workflow.
+ *
+ * @param {object} workflowToCreate - The Workflow to create.
+ */
+function create(workflowToCreate) {
```

Comment:
This code seems to be very generic. I believe that we can pull that later into some common sync implementation that would allow users to enable sync for some specific datasets.